apps/artemis-web/lib/helpers.ts

```diff
@@ -11,6 +13,41 @@ export function extractUser(req) {
   };
 }
 
+export function extractLdapUser(req) {
+  if (!req.user) return null;
+
+  const { cn, mail, employeeType } = req.user;
+
+  const user = req.db.collection('users').updateOne(
+    {
+      email: mail,
+      name: cn,
+    },
+    {
+      $set: {
+        email: mail,
+        password: '<REDUCTED>',
+        name: cn,
+        lastLogin: new Date(),
+        currentLogin: new Date(),
+        role: 'user', // just for testing. normally it will be 'pending'
```

Comment:
I don't think we ever go pending for LDAP users as they are already within the LDAP group